services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -54,6 +59,61 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
     setDisplayValues(newDisplayValues);
   };
 
+  // delete selected record from DB
+  const deleteRecord = async (selectedRecord: EntityShape) => {
+    if (userIsStateUser || userIsStateRep) {
+      const reportKeys = {
+        state: state,
+        id: report?.id,
+      };
+
+      let filteredEntities = {};
+      if (name === "plans") {
+        filteredEntities = {
+          [name]: form
+            .getValues()
+            .plans.filter((plan: AnyObject) => plan.id !== selectedRecord.id),
```

Comment:
I think you can use `[name]` here too and avoid the if/elseif. Then maybe in the filter just use a generic name